provisioner/k8s/README.md

```diff
@@ -1,3 +1,112 @@
+```dockerfile
+FROM scratch
+LABEL manifests="/manifests"
+COPY ./manifests /manifests
+```
+
+The `k8s` provisioner is able to unpack a given `k8s` bundle onto a cluster and then instantiate it, making the
+content of the bundle available in the cluster. It does so by reconciling the `Bundle` and `BundleInstance` types that 
+have the `spec.provisionerClassName` field set to `core.rukpak.io/plain`. This field must be set to the correct provisioner name 
+in order for the k8s provisioner to see and interact with the bundle. 
+
+## Use cases
+
+### Install and apply a specific version of a bundle 
+
+The k8s provisioner can install and make available a specific `k8s` bundle available in the cluster. 
+
+Simply create a `Bundle` resource pointing to a specific version of your bundle, and a `BundleInstance` which references that bundle.
+The k8s provisioner will unpack the provided Bundle onto the cluster, and eventually make the content available on the cluster. 
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle-image@sha256:xyz123
+  provisionerClassName: core.rukpak.io/plain
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/plain
+  bundleName: my-bundle
+``` 
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak APIs, `Bundle` and `BundleInstance`. 
+A user can specify a particular `Bundle` to be available in the cluster for inspection before any application of the resources. 
+Given a `Bundle` resource named `my-bundle`, the k8s provisioner will pull down and unpack the bundle to a variety of storage backends, such as ConfigMaps or PersistentVolumes.
+
+Currently the ConfigMap backend is supported by the provisioner, and the content can be inspected by viewing the metadata ConfigMap, which holds
+information on all ConfigMaps related to this bundle. 
+
+`kubectl -n rukpak-system get cm --selector core.rukpak.io/owner-name=<bundle-name>` gets all ConfigMaps related to the `Bundle` on-cluster.
+
+Surfacing the content of a bundle in a more user-friendly way, via a plugin or additional API, is on the RukPak roadmap. 
+
+### Pivot between bundle versions 
+
+The `BundleInstance` API is meant to indicate the version of the bundle that should be active within the cluster. Given two unpacked bundles in the cluster,
+`my-bundle-v0.0.1` and `my-bundle-v0.0.2`, the `spec.bundleName` field of the related `BundleInstance` can be updated to pivot to a different version. For example by pivoting from an older version,
+to the newer version, by updating the `spec.bundleName` field the k8s provisioner will create the new bundle content on-cluster and remove the old content.  
+
+## Running locally 
```

Comment:
It would nice to see an e2e workflow of creating the Bundle image from scratch from a local directory of static YAML manifests, pushing that built image to a container registry, and then creating the requisite Bundle and BundleInstance resources (but this can be done as a follow-up as well).

Reply:
Yes, now that we have the repository structure cleaned up and testing locally is easier, the local example can be expanded.

Reply:
We could use the process that I did on Combo here, if it is valuable.

Reply:
Random side note (and not an immediate concern) that I'm reminded of (and I think this is touched on in the strawman): It would be nice if the bundle API allowed for a configurable source (e.g. image vs configmap vs volume vs s3, etc.).

It seems like we might want a `source` union type that's currently just populated with an underlying `image` source.
```yaml
apiVersion: rukpak.io/v1alpha1
kind: Bundle
metadata:
  name: mycompany-bundle-v1.2.3
spec:
  source:
    image:
      ref: "quay.io/mycompany/mycompany-bundle:v1.2.3"
```

Reply:
@joelanford I created the #92 so maybe we can copy the want/need for a union type in that issue?